### ISSUE TITLE
use http client

### ISSUE THREAD

**Issue by cyberhck** [M-Yankov/SWapi-CSharp#22]
If this were to use http client and if constructor could accept a instance of client (and fallback to a default one), then I'd have ability to add handlers to the client, which means I'd be able to cache all requests locally and not have to call API too often, is that a possibility you'd be willing to take a look at?

**Comment by M-Yankov** [comment]
Yes, when I have more free time I will look on to it. Meanwhile if you already checked the code and you know what should be changed, you can open a pull request with related changes. 

Also please give an code example of usage the library with new changes. It will be clearer what needs to be changed.
